pkg/config/model.go

```diff
@@ -448,6 +454,9 @@ func (c Config) DeepCopy() Config {
 	if c.Status != nil {
 		clone.Status = DeepCopy(c.Status)
 	}
+	if c.Extra != nil {
+		clone.Extra = maps.Clone(c.Extra)
```

Comment:
note this is NOT a deep copy. We cannot really deep copy, but should call it out

Reply:
hmm, is it problematic? I dont think we change that values right now, but potentially we could?

I will leave a comment for now, but maybe its better to use MergeCopy?